src/main/webapp/scripts/analyze.js

```diff
@@ -1,44 +1,23 @@
-function analyzeVideo() {
+async function analyzeVideo() {
   const videoId = document.getElementById('video-url').value.split('v=')[1];
-  const baseUrl = window.location.origin;
-  let url = new URL('/sentiment', baseUrl);
-  url.searchParams.append('video-id', videoId);
 
-  fetch(url)
+  return fetch(`/sentiment?video-id=${videoId}`)
   .then(response => {
     if (response.status >= 200 && response.status <= 299) {
       return response.json();
-    } else if (response.status >= 400 && response.status <= 499) {
-      throw InvalidVideoException(response.statusText);
     } else {
-      throw ServletFailureException(response.statusText);
+      throw new Error(response.statusText);
     }
   })
   .then(videoAnalysis => {
-    if (videoAnalysis.dataAvailable) {
-      updateSentimentContainer(videoAnalysis.score);
+    if (videoAnalysis.scoreAvailable) {
+      updateDom(videoAnalysis.score.toString(), 'sentiment-container');
     } else {
-      updateSentimentContainer("We couldn't analyze this video! Try again.")
+      updateDom('We couldn\'t analyze this video! Try again.',
```

Comment:
Telling the user to try again is letting them think that the issue will be fixed if they try again. In this case it won't. A better message would be something like "Video has no available captions or comments to analyze."